src/vs/workbench/contrib/terminal/browser/terminalQuickFixBuiltinActions.ts

```diff
@@ -102,8 +105,10 @@ export function freePort(terminalInstance?: Partial<ITerminalInstance>): ITermin
 		}
 	};
 }
+
 export function gitPushSetUpstream(): ITerminalQuickFixOptions {
 	return {
+		source: 'builtin',
 		id: 'Git Push Set Upstream',
 		commandLineMatcher: GitPushCommandLineRegex,
```

Comment:
We should move these to use regex capture groups as well where possible so it's easier to maintain.